### ISSUE TITLE
Feature Request: support event wait list when enqueueing a Kernel or Buffer Read/Write

### ISSUE THREAD

**Issue by TizianoDeMatteis** [definelicht/hlslib#33]
It would be nice to have a way to pass a list of events that need to complete before a Kernel can start executing or a Buffer operation is executed

Ref: https://www.khronos.org/registry/OpenCL//sdk/2.2/docs/man/html/clEnqueueTask.html
(this is the C API, but I assume the same applies to the C++ one)

**Comment by definelicht** [comment]
Merged with #35.
